libs/blocks/quiz-entry/quiz-entry.js

```diff
@@ -73,127 +116,150 @@ const App = ({
+      // eslint-disable-next-line no-console
+      if (debug) console.log(currentQuizState);
+      if (!debug) window.location = quizPath;
     }
-    // console.log('quizPath', quizPath);
   };
 
   const onOptionClick = (option) => () => {
-    const newState = { ...selectedCards };
+    const selected = { ...selectedCards };
+    const selectedTotal = Object.keys(selected).length;
 
-    if (Object.keys(newState).length >= maxSelections && !newState[option.options]) {
+    if (selectedTotal >= maxSelections && !selected[option.options]) {
       return;
     }
 
-    if (!newState[option.options]) {
-      newState[option.options] = true;
+    if (!selected[option.options]) {
+      selected[option.options] = true;
     } else {
-      delete newState[option.options];
+      delete selected[option.options];
     }
 
-    setSelectedCards(newState);
-    setCountOfSelectedCards(Object.keys(newState).length);
+    setSelectedCards(selected);
   };
 
+  const onMLInput = (event) => {
+    const inputValue = event.target.value;
+    setMLInputUsed(true);
+    if (inputValue.length === 0) {
+      setMLInputUsed(false);
+    }
+  };
+
+  const onMLEnter = (event) => {
+    if (event.keyCode === enterKeyCode) continueQuiz();
+  };
+
+  if (selectedQuestion) {
+    maxSelections = +selectedQuestion['max-selections'];
+  }
+
   if (!dataLoaded || !selectedQuestion) return null;
+
   return html`<div class="quiz-entry-container">
     <div class="quiz-entry-title">${quizLists.strings[selectedQuestion.questions].heading}</div>
     <div class="quiz-entry-subtitle">${quizLists.strings[selectedQuestion.questions]['sub-head']}</div>
-    ${hasMLData && html`<${mlField} cardsUsed="${cardsUsed}" onMLInput="${onMLInput}" placeholderText="${getOptionsValue('fi_code', 'title')}"/><div class="results-container"></div>`}
+    ${hasMLData && html`<${mlField} 
+      cardsUsed="${cardsUsed}" 
+      onMLInput="${onMLInput}"
+      onMLEnter="${onMLEnter}"
+      placeholderText="${getOptionsValue('fi_code', 'title')}"/><div class="results-container"></div>`}
     <div class="quiz-entry-text">${quizLists.strings[selectedQuestion.questions].text}</div>
     ${selectedQuestion.questions && html`<${GetQuizOption} 
       maxSelections=${maxSelections} 
       options=${quizData.strings[selectedQuestion.questions]}
       background=${getStringValue('icon-background-color')}
-      countSelectedCards=${countSelectedCards}
+      countSelectedCards=${selectedCards.length}
       selectedCards=${selectedCards}
       onOptionClick=${onOptionClick}
       getOptionsValue=${getOptionsValue}
       mlInputUsed=${mlInputUsed}/>`}
     <div class="quiz-button-container">
         <button 
+          disabled="${!!(!mlInputUsed && !cardsUsed)}"
           aria-label="${quizLists.strings[selectedQuestion.questions].btn}" 
           class="quiz-button" 
-          onClick=${() => { onQuizButton(); }}>
+          onClick=${() => { continueQuiz(); }}>
```

Comment:
do you want to update this like onOptionClick?

Reply:
I mean change it to onClick=${continueQuiz}

const continueQuiz = () => () => { code }

Reply:
....that's what it is set to.

Reply:
ok